### PR TITLE
fix(cloudtrail): Invalid arn partition for GovCloud

### LINKED ISSUE #8247: cloudtrail: Construct does not account for partitions
Trail construct `logAllS3DataEvents` does not account for partition.

### Reproduction Steps

```
    const trail = new cloudtrail.Trail(this, `Trail`, {
      bucket: this.cloudTrailBucket,
    });

    trail.logAllS3DataEvents();
```

### Error Log

In GovCloud CloudFormation console:

```
Value arn:aws:s3::: for DataResources.Values is invalid. 
```

### Environment

  - **CLI Version      : 1.41.0**
  - **Framework Version: 1.41.0**
  - **OS               : macOS**
  - **Language         : TypeScript**

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by cmckni3**
Use partition ref for lambda and s3 data events

Closes #8247

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 1]
```diff
@@ -333,7 +333,7 @@ export class Trail extends Resource {
    * @default false
    */
   public logAllLambdaDataEvents(options: AddEventSelectorOptions = {}) {
-    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ 'arn:aws:lambda' ], options);
+    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ `arn:${Fn.ref('AWS::Partition')}:lambda` ], options);
```
Comment (eladb) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:336]: Use `Aws.PARTITION` instead of Fn.ref

Comment (cmckni3) [reply packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts]: Done.

`Aws` is convenient for CFN pseudo parameters!

**Comment by cmckni3** [comment]
@eladb should I update tests? I can run the tests locally to see.

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 2]
```diff
@@ -333,7 +333,7 @@ export class Trail extends Resource {
    * @default false
    */
   public logAllLambdaDataEvents(options: AddEventSelectorOptions = {}) {
-    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ 'arn:aws:lambda' ], options);
+    return this.addEventSelector(DataResourceType.LAMBDA_FUNCTION, [ `arn:${Aws.PARTITION}:lambda` ], options);
```
Comment (eladb) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:336]: Prefer `this.stack.partition` instead of `Aws.PARTITION`

**Comment by cmckni3** [comment]
> Yes, please see the contribution guide for details on how to re-run the integration test

I don't see a integration tests for `logAllS3DataEvents` and `logAllLambdaDataEvents`

I have updated the `cloudtrail.test.ts` file (unit tests I believe). They pass for me but there is an unrelated failure from a previous commit on master.

**Pull request merged**